xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-webjar/src/main/webjar/lightbox.js

```diff
@@ -172,36 +172,43 @@ define('xwiki-lightbox', [
   /*
    * Make sure that the toolbar will remain open also while hovering it, not just the image.
    */
-  var keepToolbarOpenOnHover = function(image, timeout) {
-    timeout = setTimeout(function() {
-      image.popover('hide');
+  var keepToolbarOpenOnHover = function() {
+    // Hide the image popover after 3 seconds (if the mouse doesn't enter the popover).
+    var hideTimeout = setTimeout(function() {
+      $('#imagePopoverContainer').popover('hide');
     }, 3000);
-    $('.popover').on('mouseenter', function() {
-      clearTimeout(timeout);
+    // Don't hide the image popover when the mouse is over it.
+    $('#imagePopoverContainer .popover').on('mouseenter', function() {
+      clearTimeout(hideTimeout);
     });
-    $('.popover').on('mouseleave', function() {
-      image.popover('hide');
+    // Hide the image popover when the mouse leaves its area.
+    $('#imagePopoverContainer .popover').on('mouseleave', function() {
+      $(this).popover('hide');
     });
+    return hideTimeout;
   };
 
   /**
    * Assign to each selected image a toolbar popover with download and lightbox options.
    */
   var enableToolbarPopovers = function() {
-    var timeout;
-    lightboxImages.popover({
+    var hideTimeout;
+    var showTimeout;
+    // In order to place the popover at cursor location, it would be added to an element that changes its position
+    // based on mouse events.
+    var popoverContainer = $('#imagePopoverContainer');
+    popoverContainer.popover({
```

Comment:
Don't you need to update the page  objects or the functional tests? I'm thinking at the method that forces the image popover to be shown on hover.

Reply:
Forgot it, thanks! The test were not failing since after the upgrade to selenium 4 the workaround is not needed anymore.